scala/lambdas/ingest-event-aggregator/src/main/scala/uk/gov/nationalarchives/eventaggregator/Lambda.scala

```diff
@@ -0,0 +1,37 @@
+package uk.gov.nationalarchives.eventaggregator
+
+import cats.effect.unsafe.implicits.global
+import cats.effect.{IO, Ref}
+import cats.syntax.all.*
+import com.amazonaws.services.lambda.runtime.events.SQSEvent
+import com.amazonaws.services.lambda.runtime.{Context, RequestHandler}
+import pureconfig.generic.derivation.default.*
+import pureconfig.module.catseffect.syntax.*
+import pureconfig.{ConfigReader, ConfigSource}
+import uk.gov.nationalarchives.eventaggregator.Lambda.*
+import uk.gov.nationalarchives.{DADynamoDBClient, DASFNClient}
+import io.circe.generic.auto.*
+import java.net.URI
+import java.time.Instant
+import java.util.UUID
+import scala.jdk.CollectionConverters.*
+class Lambda extends RequestHandler[SQSEvent, Unit]:
+
+  private val ref: Ref[IO, Map[String, Batch]] = Ref.unsafe[IO, Map[String, Batch]](Map[String, Batch]())
+
+  given DASFNClient[IO] = DASFNClient[IO]()
+  given DADynamoDBClient[IO] = DADynamoDBClient[IO]()
+
+  override def handleRequest(input: SQSEvent, context: Context): Unit =
+    ConfigSource.default
+      .loadF[IO, Config]()
+      .flatMap { config =>
+        Aggregator[IO].aggregate(config, ref, input.getRecords.asScala.toList, context.getRemainingTimeInMillis)
+      }
+      .unsafeRunSync()
+
+object Lambda:
+
+  case class Batch(batchId: String, expires: Instant, items: Int)
```

Comment:
1. Im not 100% sure but we do have a groupId (which I think is mainly to allow retries) so changing this to `Group` would be confusing and we're creating batches based on batchId so it's internally consistent. I'll leave it and if it's confusing when it's running we can change it.
2. I'll go with `itemCount` to avoid the preposition.
3. In this instance it's an id and a location in an S3 bucket but it doesn't really matter what the item is as we're adding it to the lock table as the message which will be processed further on. I don't want to make it more specfiic as we may re-use this for other ingest sources.

Reply:
Edit for point 1. I'll change it to group.